packages/matter.js/src/cluster/ClusterBuilder.ts

```diff
@@ -0,0 +1,57 @@
+/**
+ * @license
+ * Copyright 2022-2023 Project CHIP Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { BitSchema, TypeFromPartialBitSchema } from "../schema/index.js";
```

Comment:
I see, looking for some way of expressing "these are the bits you should use from outside this directory".  I had used "internal.ts" for that previously, could go back to that, or need some other convention.

Reply:
On the other hand having local imports go to the "real" file is just some more lines but also more clarify where it comes from ... In this case it in fact is even the same File to use ... because internal.st where all stuff is exported will have the same issue

Reply:
Maybe I'm misunderstanding the issue, I thought you were saying the index.ts files are treated specially as "public" exports?  Otherwise the files are just exporting stuff that's already exported in other files.

The files referenced by the index (or whatever) file can't use it because that'd be circular but so long as it's a tree and not a graph it should be fine.

Problem I need to solve for is files that aren't loaded directly but need to be loaded in order for their contents to be registered.  The place that they register with can't load them because it'd be circular.  If I say "only access files under this tree through this one file" that solves the problem because it can perform the load in the appropriate order.

Examples include MatterModel.ts; if you reference it directly none of the clusters will be present.  Model.create() in Model.ts will have the same problem because the factory pool will be empty.  ClusterFactory in that old PR would have the same issue.

Probably should jump on Discord for this one.  LMK if you have ideas.

Reply:
If the "index.ts" in your dir have differnt purpose and also never exposed that you are rught, you can use them - and yes then maybe renaming to "internal.ts" might be a good idea.

I more talk about "when we import the other things" like BitMapSchema.ts in this case where we have an index because of "it is used to export it" we should prevent using them ... maybe could also be an idea to call them "export.ts" instead of index.ts to even more show their meaning ... WDYT ... (but in fact the ide might still propose them to use for imports)

Reply:
Yeah if index.ts is only intended to enumerate exports I'd probably rename to "exports.ts".  Then VS code devs can use `javascript.preferences.autoImportFileExcludePatterns` to disable automatic import.  Regardless the index.ts files I've added and the existing ones should probably have different names since they have almost opposite meanings.

Or we could tweak the convention so you don't use an index in the current directory or a directory higher in the tree.  That would be easy to check for in lint and avoid circular dependencies except for circular directory references which should probably be factored away anyway.

WDYT?  I can add to this PR if you want or we could address later.

Reply:
lets do that in a separate PR afterwards ... I would leave it in the dirs because of easier export config in package.json ...